packages/flutter/lib/src/scheduler/ticker.dart

```diff
@@ -12,80 +15,250 @@ import 'binding.dart';
+  ///
+  /// If this is true, then calling [scheduleTick] should succeed.
+  ///
+  /// Reasons why a tick should not be scheduled include:
+  ///
+  /// * A tick has already been scheduled for the coming frame.
+  /// * The ticker is not active ([start] has not been called).
+  /// * The ticker is not ticking, e.g. because it is [muted] (see [isTicking]).
+  bool get shouldScheduleTick => isTicking && !scheduled;
+
   void _tick(Duration timeStamp) {
     assert(isTicking);
-    assert(_animationId != null);
+    assert(scheduled);
     _animationId = null;
 
     if (_startTime == null)
       _startTime = timeStamp;
 
     _onTick(timeStamp - _startTime);
 
-    // The onTick callback may have scheduled another tick already.
-    if (isTicking && _animationId == null)
-      _scheduleTick(rescheduling: true);
+    // The onTick callback may have scheduled another tick already, for
+    // example by calling stop then start again.
+    if (shouldScheduleTick)
+      scheduleTick(rescheduling: true);
   }
 
-  void _scheduleTick({ bool rescheduling: false }) {
+  /// Schedules a tick for the next frame.
+  ///
+  /// This should only be called if [shouldScheduleTick] is true.
+  @protected
+  void scheduleTick({ bool rescheduling: false }) {
     assert(isTicking);
-    assert(_animationId == null);
+    assert(!scheduled);
+    assert(shouldScheduleTick);
     _animationId = SchedulerBinding.instance.scheduleFrameCallback(_tick, rescheduling: rescheduling);
   }
+
+  /// Cancels the frame callback that was requested by [scheduleTick], if any.
+  ///
+  /// Calling this method when no tick is [scheduled] is harmless.
+  ///
+  /// This method should not be called when [shouldScheduleTick] would return
+  /// true if no tick was scheduled.
+  @protected
+  void unscheduleTick() {
+    if (scheduled) {
+      SchedulerBinding.instance.cancelFrameCallbackWithId(_animationId);
+      _animationId = null;
+    }
+    assert(!shouldScheduleTick);
+  }
+
+  /// Makes this ticker take the state of another ticker, and disposes the other
+  /// ticker.
+  ///
+  /// This is useful if an object with a [Ticker] is given a new
+  /// [TickerProvider] but needs to maintain continuity. In particular, this
+  /// maintains the identity of the [Future] returned by the [start] function of
+  /// the original [Ticker] if the original ticker is active.
+  ///
+  /// This ticker must not be active when this method is called.
+  void absorbTicker(Ticker originalTicker) {
+    assert(!isTicking);
+    assert(_completer == null);
+    assert(_startTime == null);
+    assert(_animationId == null);
+    _completer = originalTicker._completer;
+    _startTime = originalTicker._startTime;
+    if (shouldScheduleTick)
+      scheduleTick();
+    originalTicker.dispose();
+  }
+
+  /// Release the resources used by this object. The object is no longer usable
+  /// after this method is called.
+  @mustCallSuper
+  void dispose() {
+    _completer = null;
```

Comment:
I intentionally didn't so that the class would end up in a bogus state after dispose(). I can add a comment to that effect.